### PR TITLE
Change galaxy-helm chart to work with Galaxy image built with libyaml uwsgi

### PR DESCRIPTION
**Pull request by nuwang**
xref: https://github.com/galaxyproject/galaxy/issues/8558
xref: https://github.com/CloudVE/galaxy-helm/commit/31434159c830f376b9b2cefeb779271b49121181

### CODE REVIEW

File: galaxy/values-cvmfs.yaml [hunk 1]
```diff
@@ -171,31 +171,28 @@ configs:
...
       processes: 1
       http: 0.0.0.0:8080
-      static-map: /static/style=/galaxy/server/static/style/blue
-      static-map: /static=/galaxy/server/static
-      static-map: /favicon.ico=/galaxy/server/static/favicon.ico
       pythonpath: /galaxy/server/lib
       thunder-lock: true
       manage-script-name: true
-      mount: {{.Values.ingress.path}}=galaxy.webapps.galaxy.buildapp:uwsgi_app()
       buffer-size: 16384
```
Comment (nuwang) [inline galaxy/values-cvmfs.yaml]: @almahmoud The code snippet in the configmap for converting to yaml `{{- $original := ($entry | toYaml) -}}` trips up on this line. Therefore, I made this a commandline argument to uwsgi too, but seems wrong.

Comment (almahmoud) [reply galaxy/values-cvmfs.yaml]: Fixed and made some more changes in: https://github.com/galaxyproject/galaxy-helm/pull/55/commits/dd3bd3495e9b6b26abac6a2a291180e3e4288e4e

**Comment by nuwang** [comment]
It's great that you got this working in so many scenarios, given all the complications we've run into. A new path may have opened up: https://github.com/galaxyproject/galaxy/issues/8558
We should try switching uwsgi to libyaml in the docker container, and that could end up simplifying things quite a bit. Then we'd have fewer edge cases to handle.

Ideally, we should be able to treat this all in a simple uniform way - whether or not a config file is a single string literal or a nested dict should ideally not matter since it's all valid yaml.

WRT to deleting helm keys, that seems to be supported? https://github.com/helm/helm/pull/2648

**Comment by nuwang** [comment]
@almahmoud Let's work on the libyaml portion today or tomorrow perhaps?

File: galaxy/templates/configs-galaxy.yaml [hunk 2]
```diff
@@ -16,7 +16,36 @@ data:
 {{- end }}
   {{- range $key, $entry := .Values.configs -}}
   {{- if $entry -}}
-  {{- $key | nindent 2 }}: |
-    {{- tpl (tpl $entry $) $ | nindent 4 -}}
-  {{- end }}
-  {{- end }}
+  {{- $key | nindent 4 }}: |
```
Comment (nuwang) [inline galaxy/templates/configs-galaxy.yaml]: @almahmoud Now that the libyaml change has been made: https://github.com/CloudVE/galaxy-docker-k8s/pull/11, it seems like we should convert this to the simplest version that works. Wasn't there some version you wrote that handled both but was fairly simple?

Comment (almahmoud) [reply galaxy/templates/configs-galaxy.yaml]: Done!

Comment (almahmoud) [reply galaxy/templates/configs-galaxy.yaml]: I also rebased everything into a single commit

**Pull request merged**